examples/phyast.py

```diff
@@ -0,0 +1,390 @@
+          s = "while("+self.recompile(args[0])+","
+          if len(args)==2:
+              s += self.recompile(args[1])+")"
+          else:
+              tw = "block("
+              for aa in args[1:]:
+                  s += tw
+                  tw = ","
+                  s += self.recompile(aa)
+              s += ")"
+          s += ")"
+          return s
+      elif nm == "If":
+          s = "if("+self.recompile(a.test)+","
+          if len(a.body)>1:
+              s += "block("
+              for j in range(len(a.body)):
+                  aa = a.body[j]
+                  if j > 0:
+                      s += ","
+                  if j + 1 == len(a.body):
+                      s += self.recompile(aa,allowreturn)
+                  else:
+                      s += self.recompile(aa)
+              s += ")"
+          else:
+              s += self.recompile(a.body[0],allowreturn)
+          s += ","
+          if len(a.orelse)>1:
+              s += "block("
+              for j in range(len(a.orelse)):
+                  aa = a.orelse[j]
+                  if j > 0:
+                      s += ","
+                  if j + 1 == len(a.orelse):
+                      s += self.recompile(aa,allowreturn)
+                  else:
+                      s += self.recompile(aa)
+              s += ")"
+          else:
+              s += self.recompile(a.orelse[0],allowreturn)
+          s += ")"
+          return s
+      elif nm == "Compare":
+          args = [arg for arg in ast.iter_child_nodes(a)]
+          sym = "?"
+          nn = args[1].__class__.__name__
+          if nn == "Lt":
+              sym = "<"
+          elif nn == "Gt":
+              sym = ">"
+          elif nn == "LtE":
+              sym = "<="
+          elif nn == "GtE":
+              sym = ">="
+          elif nn == "NotEq":
+              sym = "!="
+          elif nn == "Eq":
+              sym = "=="
+          return self.recompile(args[0])+sym+self.recompile(args[2])
+      elif nm == "UnaryOp":
+          args = [arg for arg in ast.iter_child_nodes(a)]
+          nm2 = args[0].__class__.__name__
+          nm3 = args[1].__class__.__name__
+          if nm2 == "USub":
+            if nm3 == "BinOp" or self.groupAggressively:
+                return "-(" + self.recompile(args[1])+")"
+            else:
+                return "-" + self.recompile(args[1])
+          else:
+            raise Exception(nm2)
+      else:
+          raise Exception(nm)
+
+local_namespace = {}
+# Create the decorator
+class phylanx(object):
+    def __init__(self,f):
+        global local_namespace
+        self.f = f
+        # Get the source code
+        src = inspect.getsource(f) #getsource(f)
+        # Before recompiling the code, take
+        # off the decorator from the source.
+        src = re.sub(r'^\s*@\w+\s*','',src)
+
+        # Create the AST
+        tree = ast.parse(src)
+        #astdump.indented(tree)
```

Comment:
Please remove code that is commented out

Reply:
Also, the decorator should probably go into the phylanx/python directory for everybody to use.

Reply:
handled.